pty/start_other.go

```diff
@@ -35,18 +34,16 @@ func startPty(cmd *exec.Cmd) (PTY, *os.Process, error) {
 			// TTY resolves it.
 			return startPty(cmd)
 		}
-		return nil, nil, xerrors.Errorf("start: %w", err)
+		return nil, xerrors.Errorf("start: %w", err)
 	}
-	go func() {
-		// The GC can garbage collect the TTY FD before the command
-		// has finished running. See:
-		// https://github.com/creack/pty/issues/127#issuecomment-932764012
-		_ = cmd.Wait()
-		runtime.KeepAlive(ptty)
-	}()
-	oPty := &otherPty{
-		pty: ptty,
-		tty: tty,
+	oPty := &otherPtyWithProcess{
+		otherPty: &otherPty{
+			pty: ptty,
+			tty: tty,
+		},
+		cmd:     cmd,
+		cmdDone: make(chan any),
```

Comment:
(non-blocking): Is there a reason to make this a `chan any` versus a `chan bool` or `chan struct{}`? Mainly curious.

Reply:
A motivation for using `chan struct{}` is that `struct{}` is the smallest data type in Go (zero).

The savings would be quite insignificant here, but it's a common practice. It also signifies we're not interested in what type of values are passed through it.

Reply:
In this case it doesn't matter what kind of channel it is, since we only use it to synchronize things waiting for the command to complete by closing it.

For a large, buffered channel you could make the case for one or other of these on the grounds that they have different memory storage requirements.